src/display/canvas.js

```diff
@@ -218,7 +212,7 @@ var CachedCanvases = (function CachedCanvasesClosure() {
         // reset canvas transform for emulated mozCurrentTransform, if needed
         canvasEntry.context.setTransform(1, 0, 0, 1, 0, 0);
       } else {
-        var canvas = createScratchCanvas(width, height);
+        var canvas = CanvasGraphics.DOMcanvasFactory.create(width, height);
```

Comment:
In order to prevent errors, such as `TypeError: CanvasGraphics.DOMcanvasFactory is undefined[Learn More]  canvas.js:215:13`, you'll need to pass in a `canvasFactory` instance when initializing `CachedCanvases`, i.e. change the "constructor" to 
```javascript
function CachedCanvases(canvasFactory) {
     this.canvasFactory = this.canvasFactory;
     this.cache = Object.create(null);
   }
}
```
and then use `this.canvasFactory.create(width, height)` here.